perma_web/fabfile/dev.py

```diff
@@ -74,7 +76,16 @@ def test_python(apps=_default_tests):
     if "functional_tests" in apps and not os.environ.get('SERVER_URL'):
         local("DJANGO__STATICFILES_STORAGE=django.contrib.staticfiles.storage.StaticFilesStorage python manage.py collectstatic --noinput")
 
-    local("coverage run manage.py test --settings perma.settings.deployments.settings_testing %s" % (apps))
+    # temporarily set MEDIA_ROOT to a tmp directory, in a way that lets us clean up after ourselves
```

Comment:
You can use the `tempdir` package (which we also use in tasks.py) to simplify this a bit:

```
import tempdir

with tempdir.in_tempdir() as tmp:
    shell_envs = {
        'DJANGO__MEDIA_ROOT': tmp.name
    }
    local(...)
```

This takes care of wrapping in a try-finally, which otherwise you'd want to do here.

Reply:
Aha! That's much nicer, thanks.